nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/processors/azure/cosmos/document/ITGetCosmosDocumentRecord.java

```diff
@@ -0,0 +1,164 @@
+    private static final String TEST_PARTITION_KEY = "category";
+    private static List<JsonNode> testData;
+    private static int numOfTestData = 10;
+    private static RecordSchema SCHEMA;
+
+    static {
+        final ObjectMapper mapper = new ObjectMapper();
+        final List<RecordField> testDataFields = new ArrayList<>();
+        final RecordField idField = new RecordField("id", RecordFieldType.STRING.getDataType());
+        final RecordField categoryField = new RecordField(TEST_PARTITION_KEY, RecordFieldType.INT.getDataType());
+        final RecordField payloadField = new RecordField("payload", RecordFieldType.STRING.getDataType());
+        testDataFields.add(idField);
+        testDataFields.add(categoryField);
+        testDataFields.add(payloadField);;
+        SCHEMA = new SimpleRecordSchema(testDataFields);
+        JsonNode doc = null;
+
+        testData = new ArrayList<>();
+        for (int i=0; i< numOfTestData; i++) {
+            JsonObject json =  new JsonObject();
+            json.addProperty("id", ""+i);
+            json.addProperty(TEST_COSMOS_PARTITION_KEY_FIELD_NAME, MockTestBase.getRandomInt(1,4));
+            json.addProperty("payload", RandomStringUtils.random(100, true, true));
+            try {
+                doc = mapper.readTree(json.toString());
+            } catch(IOException exp) {
+                exp.printStackTrace();
+            }
+            testData.add(doc);
+        }
+
+        for (JsonNode jdoc : testData) {
+            try {
+                container.upsertItem(jdoc);
+            } catch(CosmosException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+    @Override
+    protected Class<? extends Processor> getProcessorClass() {
+        return GetAzureCosmosDBRecord.class;
+    }
+    @Before
+    public void setupTest() throws Exception {
+        runner.setProperty(GetAzureCosmosDBRecord.QUERY, TEST_COSMOS_QUERY);
+        MockSchemaRegistry registry = new MockSchemaRegistry();
+        JsonRecordSetWriter writer = new JsonRecordSetWriter();
+        registry.addSchema("sample", SCHEMA);
+
+        runner.addControllerService("writer", writer);
+        runner.addControllerService("registry", registry);
+        runner.setProperty(writer, SchemaAccessUtils.SCHEMA_REGISTRY, "registry");
+        runner.setProperty(writer, SchemaAccessUtils.SCHEMA_ACCESS_STRATEGY, SchemaAccessUtils.SCHEMA_NAME_PROPERTY);
+        runner.enableControllerService(registry);
+        runner.enableControllerService(writer);
+        runner.setProperty(GetAzureCosmosDBRecord.WRITER_FACTORY, "writer");
+    }
+    @Test
+    public void testReadRecords() {
+        runner.setVariable("schema.name", "sample");
+        runner.enqueue(new byte[] {});
+        runner.run();
+
+        List<MockFlowFile> flowFiles = runner.getFlowFilesForRelationship(AbstractAzureCosmosDBProcessor.REL_SUCCESS);
+        assertTrue(flowFiles.size() == 1);
```

Comment:
Optionally, you can possibly use assertEquals here and also in the other test cases.